src/game/cGangs.cpp

```diff
@@ -492,16 +489,24 @@ void sGang::BoostSkill(int* affect_skill, int count)
     *    ex. 2: 30 points in skill. (70/30)^2 = 5.44. Possible point range: 2.72 to 8.16
     *    ex. 3: 75 points in skill. (70/75)^2 = 0.87. Possible point range: 0.44 to 1.31
     */
+
+    auto sqr = [](double x) { return x*x; };
+
     for (int j = 0; j < count; j++)  // we'll boost the skill/stat "count" number of times
     {
         if (*affect_skill < 1) *affect_skill = 1;
 
-        double boost_amount = pow(70 / (double)*affect_skill, 2);
+        double boost_amount = sqr(70 / (double)*affect_skill);
         if (boost_amount > 5) boost_amount = 5;
 
-        boost_amount = (double)g_Dice.in_range(int((boost_amount / 2) * 100), int((boost_amount*1.5) * 100)) / 100;
-        char one_more = g_Dice.percent(int((boost_amount - (int)boost_amount) * 100)) ? 1 : 0;
-        char final_boost = (char)boost_amount + one_more;
+        std::uniform_real_distribution<>
+           boost_dist{boost_amount * 0.5, boost_amount * 1.5};
```

Comment:
A floating-point `flat()` would be useful here.

Reply:
a general random utility should definitely have a function to return random floats.